src/Toolkit/Collection.php

```diff
@@ -553,7 +553,7 @@ public function group(
 				}
 			}
 
-			return new self($groups);
+			return new self($groups, !$caseInsensitive);
```

Comment:
```suggestion
			return new self($groups, $caseInsensitive);
```

Reply:
It should be without `!` to respect parent `$caseInsensitive`.

Reply:
@afbora no it's correct, the method and class use opposites: caseSensitive vs caseInsensitive

Reply:
Oh sorry, I thought both are same variable names 😅